### PR TITLE
Splits module configs into local, private, and consensus files

### LINKED ISSUE #960: Split configs into keys, local, and consensus
We should split the existing config file into 3 different files:
- keys.conf - encrypted private key material
- local.conf - editable by the user
- consensus.conf - controls the consensus config (will need another PR to use only during consensus upgrades)

We should consider removing unnecessary config values from the command-line config gen (denominations, federation_name, etc) and instead have users edit the relevant files (or provide a generic way to pass in changes via command line).

### PR DESCRIPTION
**Pull request by jkitman**
Final splitting of the module configs into local, private, and consensus files.

Fix #960

### CODE REVIEW

**Review by elsirion — APPROVED** [review]
LGTM

File: fedimint-server/src/config.rs [hunk 1]
```diff
@@ -41,41 +41,57 @@ pub struct ServerConfig {
...
     pub hbbft_bind_addr: String,
+    /// Our bind address for our API endpoints
     pub api_bind_addr: String,
+    /// Our publicly known TLS cert
     #[serde(with = "serde_tls_cert")]
     pub tls_cert: rustls::Certificate,
+    /// How many API connections we will accept
     pub max_connections: u32,
+    /// Non-consensus, non-private configuration from modules
     pub modules: BTreeMap<String, serde_json::Value>,
```
Comment (justinmoon) [inline fedimint-server/src/config.rs:95]: Nice comments

**Comment by dpc** [comment]
@jkitman It came to our attention that connection strings are currently part of the consensus config.

IMO they shouldn't . Fundamentally they are not what the consensus agrees on, only a config that allows federation to connect to each other. Depending on network setup not everyone might even be able to use the same connection strings (people running stuff behind proxies, network firewalls, VPNs, ssh forwarding etc.).

**Comment by jkitman** [comment]
@dpc Sure, so long as the TLS certs are in consensus as the single source of server-to-server trust that should be sufficient for gaurdians to safely connect to the federation.

The client-server endpoints should be in consensus IMO since client configs need to be derivable from consensus.  This has useful purposes such as determining whether a update to the client configs should be trusted (e.g. they could be signed by a trusted threshold pubkey).  Once these are Tor-based network inconsistencies will be less of a problem.

I mentioned in #1150 long-term we should use some kind of P2P DNS to simplify the sharing and updating of connection strings, perhaps for both the client-server and server-server endpoints.

**Comment by dpc** [comment]
@jkitman 

> Sure, so long as the TLS certs are in consensus as the single source of server-to-server trust that should be sufficient for gaurdians to safely connect to the federation.

I'm not if TLS certs need to be part of the consensus directly. It seems to me that each peer should have a liberty of rotating their TLS certs. What if their CA was compromised and now they need to rotate due to no fault of their own etc.?

Some public key for each member should be part of the federation (possibly one from #754 ) and then TLS certs be signed (and anything else in control of each guardian) by the public key, to afford a degree of flexibility here.


> The client-server endpoints should be in consensus IMO since client configs need to be derivable from consensus. 

Possibly yes. What exactly do we mean by "client-server endpoints"? The connection string?

**Pull request merged**